lightning-invoice/src/utils.rs

```diff
@@ -218,6 +198,85 @@ where
 	}
 }
 
+/// Utility to filter channels for an invoice, and return the corresponding route hints to include
+/// in the invoice.
+///
+/// The filtering ensures that the RouteHints returned only include the highest inbound capacity
+/// channel per counterparty node for which channels are present. Channels with a lower inbound
+/// capacity than the invoice amount aren't included in the hints. If any of the channels are
+/// private, the function returns no RouteHints, and the sender will need to look at the public
+/// channels to find the path instead.
+///
+/// Input:
+/// `channels`: The channels to filter.
+/// `min_inbound_capacity_required`: Defines the lowest inbound capacity a channel must have to not
+/// be filtered out.
+///
+/// Result:
+/// `Vec<RouteHint>`: The filtered RouteHints
+/// `bool`: True if any of the inputted channels are private, else false
+fn filter_channels(channels: Vec<ChannelDetails>, min_inbound_capacity_required: Option<u64>) -> (Vec<RouteHint>, bool){
```

Comment:
Would prefer to name this `min_inbound_capacity_msat`.

Reply:
Ok makes sense!